python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -487,35 +478,25 @@ def get_partition_names(
 
 def get_partition_tags(
     repo_def: RepositoryDefinition,
-    partition_set_name: str,
+    job_name: str,
     partition_name: str,
+    selected_asset_keys: Optional[AbstractSet[AssetKey]],
     instance_ref: Optional[InstanceRef] = None,
-):
+) -> Union[ExternalPartitionTagsData, ExternalPartitionExecutionErrorData]:
     try:
-        (
-            job_def,
-            partitions_def,
-            partitioned_config,
-        ) = _get_job_partitions_and_config_for_partition_set_name(repo_def, partition_set_name)
-
-        # Certain gRPC servers do not have access to the instance, so we only attempt to instantiate
-        # the instance when necessary for dynamic partitions: https://github.com/dagster-io/dagster/issues/12440
+        job_def = repo_def.get_job(job_name)
 
-        with _instance_from_ref_for_dynamic_partitions(instance_ref, partitions_def) as instance:
-            with user_code_error_boundary(
-                PartitionExecutionError,
-                lambda: (
-                    "Error occurred during the evaluation of the `tags_for_partition` function for"
-                    f" partitioned config on job '{job_def.name}'"
-                ),
-            ):
-                partitions_def.validate_partition_key(
-                    partition_name, dynamic_partitions_store=instance
-                )
```

Comment:
was removing this part intentional / related to this PR?

Reply:
It was intentional, though I should have flagged it and could be easily nudged to break it out into its own PR.

If you look at `_instance_from_ref_for_dynamic_partitions`, there's this weirdness right now where we only construct an instance ref if the `PartitionsDefinition` is a `DynamicPartitionsDefinition`, because otherwise we want the user to not need a dagster.yaml in their code server.

HOWEVER, part of what's happening in this PR is that we abstract away the `PartitionsDefinition` from this code path (this allows a future where someone executes a run for "2024-04-23" for asset1 and asset2, but those assets have different daily `PartitionsDefinition`s with different start dates). So now we don't have the `PartitionsDefinition` to look at and see if it's dynamic.

We could add some sort of `JobDefinition.partitions_def_is_dynamic(selected_asset_keys)` method to retain the existing behavior. HOWEVER HOWEVER, the reason I deleted this code is that I question whether this validation is really important. It result in an extra round trip to the DB. And this weirdness with the instance ref. And the backend code paths that invoke it already know the partition exists.